core/ops_builtin_v8.rs

```diff
@@ -35,6 +35,69 @@ pub fn op_unref_op(scope: &mut v8::HandleScope, promise_id: i32) {
   context_state.borrow_mut().unrefed_ops.insert(promise_id);
 }
 
+#[op2(reentrant)]
+#[global]
+pub fn op_lazy_load_esm(
+  scope: &mut v8::HandleScope,
+  #[string] module_specifier: String,
+) -> Result<v8::Global<v8::Value>, Error> {
+  let module_map_rc = JsRealm::module_map_from(scope);
+
+  let lazy_esm = module_map_rc.data.borrow().lazy_esm.clone();
+  let loader = crate::modules::LazyEsmModuleLoader::new(lazy_esm);
+
+  let mod_ns = futures::executor::block_on(async {
+    let specifier = crate::ModuleSpecifier::parse(&module_specifier)?;
```

Comment:
Factor out most of this function to a helper method on the `ModuleMap`. Also currently it's not idempotent and I think it will crash it once tries to load the same module twice.